lib/cdo/rack/cookie_dcdo.rb

```diff
@@ -0,0 +1,32 @@
+require 'json'
+require 'request_store'
+require 'dynamic_config/dcdo'
+
+# This middleware enables the setting of DCDO via cookies for testing purposes.
+# See: 'dashboard/test/ui/features/cookie_dcdo.rb'
+module Rack
+  class CookieDCDO
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      # Stores the cookie DCDO data per request.
+      RequestStore.store[:DCDO] = JSON.parse(Rack::Request.new(env).cookies['DCDO'] || '{}')
+
+      unless DCDO.instance_variable_get(:@_redefined)
+        # Redefines `DCDO#get` to return the cookie DCDO value if it exists.
+        DCDO.define_singleton_method(:get) do |key, *args|
+          RequestStore.store[:DCDO]&.key?(key) ? RequestStore.store[:DCDO][key] : super(key, *args)
+        end
+
+        DCDO.instance_variable_set(:@_redefined, true)
+      end
```

Comment:
A Rack middleware definitely doesn't feel like the best place to implement this modification to DCDO functionality. Any reason not to just put this logic in the class itself, gated by an `if CDO.cookie_dcdo` check?

Reply:
I usually prefer to avoid metaprogramming, but in this case, I have two reasons for doing so:
1. Since this is intended for testing purposes only, I didn't want to modify the original method, as that would burden it with redundant checks in production and other environments.
2. It's easier to understand the business logic when everything is kept together in one place.